mne/io/neuralynx/neuralynx.py

```diff
@@ -0,0 +1,85 @@
+import os.path as op
+from mne.io import BaseRaw
+from mne.io.utils import _read_segments_file
+from mne import create_info
+from mne.utils import logger, verbose, fill_doc
+
+
+@fill_doc
+def read_raw_neuralynx(fname, preload=False, verbose=None):
+    """Reader for an Neuralynx files.
+
+    Parameters
+    ----------
+    fname : path-like
+        Path to a folder with Neuralynx .ncs files
+    %(preload)s
+    %(verbose)s
+
+    Returns
+    -------
+    raw : instance of RawNeuralynx
+        A Raw object containing Neuralynx data.
+        See :class:`mne.io.Raw` for documentation of attributes and methods.
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attributes and methods of RawNeuralynx.
+    """
+    return RawNeuralynx(fname, preload, verbose)
+
+
+@fill_doc
+class RawNeuralynx(BaseRaw):
+    @verbose
+    def __init__(self, fname, *, preload=False, verbose=None):
+        try:
+            from neo.io import NeuralynxIO
+        except Exception:
+            raise ImportError("Missing the neo-python package") from None
```

Comment:
We have helpers for a lot of common patterns like this one
```suggestion
        neo_io = _soft_import("neo.io", "reading Neuralynx data", strict=True)
        NeuralynxIO = neo.io.NeuralynxIO
```

Reply:
last line should be

```py
NeuralynxIO = neo_io.NeuralynxIO
```
(neo_io underscore, not neo.io dot)